src/main/scala/org/mauritania/main4ino/config/Loadable.scala

```diff
@@ -1,20 +1,23 @@
 package org.mauritania.main4ino.config
 
-import cats.effect.IO
+import java.io.File
+
+import cats.effect.Sync
 import com.typesafe.config.ConfigFactory
 import pureconfig.error.ConfigReaderException
 
 import scala.reflect.ClassTag
 
-trait Loadable {
+object Loadable {
```

Comment:
@mebubo this is what I was talking about yesterday.

Reply:
Compare the way `Filesystem` and `FilesystemSync` is implemented (a part in `Algebras` and a part in `Modules`), and the way `Loadable` is (all in an object), as the required `implicit reader` seems to dirty to be put in an algebra. Any thoughts?

Reply:
@generoso ?

Reply:
^^^^^ What do you think?

Reply:
@mebubo do you mean something like this?: 
```
import pureconfig.error.ConfigReaderException

import scala.reflect.ClassTag

trait Loadable[F[_], T] {
  def loadFromFile(configFile: File): F[T]
}

class LoadableIO[T: ClassTag](implicit val D: Derivation[ConfigReader[T]]) extends Loadable[IO, T] {
  import pureconfig._

  def loadFromFile(configFile: File): IO[T] = {
    val f = configFile.getAbsoluteFile
    implicitly[Sync[IO]].fromEither {
      loadConfig[T](ConfigFactory.parseFile(f))
        .swap
        .map(i => new IllegalArgumentException(s"Cannot find/parse file '$f'", new ConfigReaderException[T](i)))
        .swap
    }
  }
}
```
However there is a class to be created per T (the configuration class want to decode). Any other ideas?

Reply:
Sorry, I didn't see this comment and struggling to remember what this was about.

I think what I was saying is that reader does not have to be an implicit argument, it can be explicit like 

```
class Loadable(val reader: ...) ...
```

The questions about having a class per T seems to be about codecs, so not really related?